File: palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        ListNode* temp=head;
        int c=0;
        stack<int>st;
        while(temp){
            st.push(temp->val);
            c++;
            temp=temp->next;
        }
        c/=2;
        temp=head;
        while(c){
            if(temp->val!=st.top())
            return false;
            st.pop();
            temp=temp->next;
            c--;
        }
        return true;
    }
};

File: leetcode/palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        lst = []

        cur = head

        while cur:
            lst.append(cur.val)
            cur = cur.next
        a = lst[::-1]
        print(lst,a)
        if lst == a:
            return True
        else:
            return False
        